### DISCUSSION TITLE
example/demo/tuto using maplibre-native as library in C++ application

### DISCUSSION THREAD

**Discussion by grabouillon06** [maplibre/maplibre-native#1160] [Q&A] [answered]
Hello,

Some time ago, I've implemented a WEB server-client mapping project using Mapbox GL JS in order to animate multiple POIs on a map. I would like to migrate this project on my Ubuntu Linux platform, meaning a C++ application.

For that:
- I'm migrating from mapbox to maplibre-native
- I've built the maplibre-native on Linux and tested the ./build/bin/mbgl-render generating the .png file -> it worked
- I think I'm ready to build my C++ app on top of the maplibre-native library

Before digging more deeply into the code of the library (and its API), Is there any example/demo/tuto or document that shows basics about the way to create such C++ application, that would help me to get the overall picture? How to instantiate the map, add basic markers, render the map ...

your help will be very much appreciated.

Thanks,

Have a good day,

br,

Olivier

**Comment by donnyv** [comment, marked as answer]
Look at the tests folder. Should have everything you need to get started.
https://github.com/maplibre/maplibre-native/tree/main/test

Some map loading tests
https://github.com/maplibre/maplibre-native/tree/main/test/map
